src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -812,11 +818,31 @@ def elastic_pool_list_capabilities(
 #                sql server                   #
 ###############################################
 
+def server_create(
+        client,
+        resource_group_name,
+        server_name,
+        assign_identity=False,
+        **kwargs):
+
+    if assign_identity is True:
+        kwargs['identity'] = ResourceIdentity(type="SystemAssigned")
```

Comment:
use enum: `IdentityType.system_assigned.value`

Reply:
see https://github.com/Azure/azure-sdk-for-python/blob/c0713004be3670ec30cdfb7e16a2978b95f455c1/azure-mgmt-sql/azure/mgmt/sql/models/sql_management_client_enums.py#L303